ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.blobs;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlobSidecar;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.datastructures.blocks.SlotAndBlockRoot;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.deneb.BeaconBlockBodyDeneb;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.BlobIdentifier;
+
+public class BlockBlobSidecarsTracker {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final SlotAndBlockRoot slotAndBlockRoot;
+
+  private final AtomicReference<Optional<BeaconBlockBodyDeneb>> blockBody =
+      new AtomicReference<>(Optional.empty());
+
+  private final Map<UInt64, BlobSidecar> blobSidecars = new ConcurrentHashMap<>();
+  private final SafeFuture<Void> blobSidecarsComplete = new SafeFuture<>();
+
+  public BlockBlobSidecarsTracker(final SlotAndBlockRoot slotAndBlockRoot) {
+    this.slotAndBlockRoot = slotAndBlockRoot;
+  }
+
+  public Map<UInt64, BlobSidecar> getBlobSidecars() {
```

Comment:
no because the plan is to use method in the availability checker to get current blobSidecars and perform the batch check on those while waiting for completion.